capstone/src/main/java/com/google/sps/servlets/MarkCheckboxServlet.java

```diff
@@ -21,23 +21,24 @@
 
 import error.ErrorHandler;
 
-@WebServlet("/update-votes")
-public class UpdateVotesServlet extends HttpServlet {
+@WebServlet("/mark-checkbox")
+public class MarkCheckboxServlet extends HttpServlet {
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Get id of changed checkbox
-    String optionIdString = request.getParameter("id");
-    boolean isOptionChecked = Boolean.parseBoolean(request.getParameter("checked"));
-    long optionId = this.parseToLong(response, optionIdString);
+    String idString = request.getParameter("id");
+    boolean isChecked = Boolean.parseBoolean(request.getParameter("checked"));
+    // type representing whether checkbox is for Option or for Challenge
+    String type = request.getParameter("type");
+    long id = this.parseToLong(response, idString);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     String userId = this.getUserId(response);
-    Entity optionEntity = this.getEntityFromId(response, optionId, datastore);
-    Set<String> votesSet = this.getUpdatedVotes(optionEntity, isOptionChecked, userId);
+    Entity entity = this.getEntityFromId(response, id, datastore, type);
+    Set<String> votesSet = this.getUpdatedVotes(entity, isChecked, userId);
 
     // Update datastore
-    optionEntity.setProperty("votes", votesSet);
-    datastore.put(optionEntity);
+    entity.setProperty("votes", votesSet);
+    datastore.put(entity);
```

Comment:
I don't see any error handling in the above lines of code, and I do see that any time you call doPost, that you're updating the database.

I would think about this more defensively: If somebody wanted to attack this endpoint, what could they send? How do I invalidate an 'attack'?

Reply:
Some portion of this is directed more broadly:

@anikabagga  @himaniyadav the three of you may want to create an abstract class AuthenticatedServlet which would override doPost and doGet and would check things like:
- a user is logged in and has a valid ID
- (Any other checks you would want for ALL gets/posts)

then you could have something like:

```java
class AuthenticatedServlet extends HttpServlet {
  ...
}

class MarkCheckboxServlet extends AuthenticatedServlet {
  @Override
  public void doPost(String userId, HttpServletRequest request, HttpServletResponse response) throws IOException {
    ...
  }
}
```

Where you can guarantee that the logic you're writing is only called when a valid userId was found